lib/utils/fs_test.go

```diff
@@ -26,6 +26,80 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestOpenFileSymlinks(t *testing.T) {
+	// symlink structure setup, this will produce the following structure under the temp directory created below:
+	// dir
+	// dir-s -> dir
+	// dir-s-s -> dir-s
+	// dir/file
+	// dir/file-s -> dir/file
+	// circular-s -> circular-s
+	// broken-s -> nonexistent
+	rootDir, err := os.MkdirTemp("", "symlink-test")
+	require.NoError(t, err)
+	defer os.RemoveAll(rootDir)
```

Comment:
There is a handy utility, which handles cleanup too:
```suggestion
	rootDir := t.TempDir()
```